wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -268,7 +283,13 @@ private[wallet] trait TransactionProcessing extends WalletLogger {
         updatedF.map(Some(_))
       case TxoState.Reserved | TxoState.ConfirmedSpent |
```

Comment:
So this isn't making much sense to me. I think you are including too many states here. 

If we ALREADY have a utxo that is `TxoState.ConfirmedSpent` you want to re-write the ALREADY EXISTING `spendingTxId` with a new one? 

This is getting into reorg territory I believe (#2179), but we need to think through these state transitions and test them more. 

I think the valid `TxoState` for this pattern match is just `TxoState.Reserved`? I'm not exactly sure how to think about `TxoState.DoesNotExist`, as that indicates that our wallet was unaware of one of _our own_ utxos to begin with?

Reply:
> If we ALREADY have a utxo that is TxoState.ConfirmedSpent you want to re-write the ALREADY EXISTING spendingTxId with a new one?
> This is getting into reorg territory I believe (#2179), but we need to think through these state transitions and test them more.

Yeah these would occur in a RBF reorg scenario.

> I'm not exactly sure how to think about TxoState.DoesNotExist, as that indicates that our wallet was unaware of one of our own utxos to begin with?

Yeah, this is kinda hard to decide because we don't really have a definition of what this state represents. It's only used when we RBF utxos